spec/controllers/api/orders_controller_spec.rb

```diff
@@ -0,0 +1,160 @@
+require 'spec_helper'
+require 'spree/api/testing_support/helpers'
+
+module Api
+  describe OrdersController, type: :controller do
+    include AuthenticationWorkflow
+    render_views
+
+    describe '#index' do
+      let!(:distributor) { create(:distributor_enterprise) }
+      let!(:distributor2) { create(:distributor_enterprise) }
+      let!(:supplier) { create(:supplier_enterprise) }
+      let!(:coordinator) { create(:distributor_enterprise) }
+      let!(:order_cycle) { create(:simple_order_cycle, coordinator: coordinator) }
+      let!(:order1) do
+        create(:order, order_cycle: order_cycle, state: 'complete', completed_at: Time.zone.now,
+                       distributor: distributor, billing_address: create(:address) )
+      end
+      let!(:order2) do
+        create(:order, order_cycle: order_cycle, state: 'complete', completed_at: Time.zone.now,
+                       distributor: distributor2, billing_address: create(:address) )
+      end
+      let!(:order3) do
+        create(:order, order_cycle: order_cycle, state: 'complete', completed_at: Time.zone.now,
+                       distributor: distributor, billing_address: create(:address) )
+      end
+      let!(:line_item1) do
+        create(:line_item, order: order1,
+                           product: create(:product, supplier: supplier))
+      end
+      let!(:line_item2) do
+        create(:line_item, order: order2,
+                           product: create(:product, supplier: supplier))
+      end
+      let!(:line_item3) do
+        create(:line_item, order: order2,
+                           product: create(:product, supplier: supplier))
+      end
+      let!(:line_item4) do
+        create(:line_item, order: order3,
+                           product: create(:product, supplier: supplier))
+      end
+      let!(:regular_user) { create(:user) }
+      let!(:admin_user) { create(:admin_user) }
+
+      context 'as a regular user' do
+        before do
+          allow(controller).to receive(:spree_current_user) { regular_user }
+          get :index
+        end
+
+        it "returns unauthorized" do
+          assert_unauthorized!
+        end
+      end
+
+      context 'as an admin user' do
+        before do
+          allow(controller).to receive(:spree_current_user) { admin_user }
+          get :index
+        end
+
+        it "retrieves a list of orders with appropriate attributes,
+            including line items with appropriate attributes" do
+
+          returns_orders(json_response)
+        end
+
+        it "formats completed_at to 'yyyy-mm-dd hh:mm'" do
+          completed_dates = json_response['orders'].map{ |order| order['completed_at'] }
```

Comment:
We have [json_spec](https://github.com/collectiveidea/json_spec) in the Gemfile, which might be of help in this spec file.